homeassistant/components/matter/entity.py

```diff
@@ -101,37 +103,45 @@ def __init__(
             identifiers={(DOMAIN, f"{ID_TYPE_DEVICE_ID}_{node_device_id}")}
         )
         self._attr_available = self._endpoint.node.available
-        # mark endpoint postfix if the device has the primary attribute on multiple endpoints
+        # mark endpoint postfix if the device has the primary attribute's cluster on multiple endpoints
         if not self._endpoint.node.is_bridge_device and any(
             ep
             for ep in self._endpoint.node.endpoints.values()
             if ep != self._endpoint
-            and ep.has_attribute(None, entity_info.primary_attribute)
+            and ep.has_cluster(entity_info.primary_attribute.cluster_id)
         ):
             self._name_postfix = str(self._endpoint.endpoint_id)
             if self._platform_translation_key and not self.translation_key:
                 self._attr_translation_key = self._platform_translation_key
 
-        # prefer the label attribute for the entity name
         # Matter has a way for users and/or vendors to specify a name for an endpoint
-        # which is always preferred over a standard HA (generated) name
-        for attr in (
-            clusters.FixedLabel.Attributes.LabelList,
-            clusters.UserLabel.Attributes.LabelList,
-        ):
-            if not (labels := self.get_matter_attribute_value(attr)):
-                continue
-            for label in labels:
-                if label.label not in ["Label", "Button"]:
+        # which is preferred over a standard HA (generated) name when the
+        # entity request it by including a list of labels that may be used for naming
+        if entity_info.entity_description.name_using_matter_labels is not None:
+            for attr in (
+                clusters.FixedLabel.Attributes.LabelList,
+                clusters.UserLabel.Attributes.LabelList,
```

Comment:
should we perhaps swap userlabel and fixedlabel so we prefer a userlabel over a fixedlabel ?

Reply:
I believe the way I implemented this, the order of preference will be set by the order in name_using_matter_labels.

In anycase, I think FixedLabel is the better default as the reason for the label is to give the end-user the manufacturer specified "hint" as to what the endpoint does. The user can then rename it as they please. 

The "UserLabel" is really more like the renaming. If anything, I was inclined to eliminate the "UserLabel", but I kept it there as it was there in the prior implementation, so I was assuming it was thought to be useful.